packages/orchestration/src/exos/chain-hub.js

```diff
@@ -54,7 +54,38 @@ export const connectionKey = (chainId1, chainId2) => {
   return [chainId1, chainId2].sort().join(CHAIN_ID_SEPARATOR);
 };
 
+const ChainIdArgShape = M.or(
+  M.string(),
+  M.splitRecord(
+    {
+      chainId: M.string(),
+    },
+    undefined,
+    M.any(),
+  ),
+);
+
+const ChainHubI = M.interface('ChainHub', {
+  registerChain: M.call(M.string(), CosmosChainInfoShape).returns(),
+  getChainInfo: M.callWhen(M.string()).returns(CosmosChainInfoShape),
+  registerConnection: M.callWhen(
+    M.string(),
+    M.string(),
+    IBCConnectionInfoShape,
+  ).returns(),
+  getConnectionInfo: M.callWhen(ChainIdArgShape, ChainIdArgShape).returns(
```

Comment:
`callWhen` is OK in asyncFlow after all?

I can imagine waiting for tests that show that it's not, but in that case, why not wait for tests that show that chainHub has to be an exo in the 1st place?

Reply:
Nope, I learned last night it's not.